### ISSUE TITLE
[BugFix]  Fix problem with class loading used by scalafmt 

### ISSUE THREAD

**Issue by marcin-jozefowicz** [softwaremill/adopt-tapir#124]
# Problem
In our application we are using `scalafmt` as standalone library: https://scalameta.org/scalafmt/docs/installation.html#standalone-library 

It dynamically downloads all necessary dependencies at runtime. It occurs that sometimes when downloading the required jobs fails, app cannot generate zip file, and throw 500 Internal Error.

```shell
2022-08-11T10:40:58.751037523Zerror: /tmp/sbtProject9026876679681506486/.scalafmt.conf: org.scalafmt.interfaces.ScalafmtException: [v3.5.8] failed to download Caused by: coursierapi.error.DownloadingArtifactsError: https://repo1.maven.org/maven2/org/scala-lang/scala-compiler/2.13.8/scala-compiler-2.13.8.jar: download error: Caught java.net.SocketException: Connection reset (Connection reset) while downloading https://repo1.maven.org/maven2/org/scala-lang/scala-compiler/2.13.8/scala-compiler-2.13.8.jar at coursierapi.error.DownloadingArtifactsError.of(DownloadingArtifactsError.java:23) at coursierapi.shaded.coursier.internal.api.ApiHelper$.doFetch(ApiHelper.scala:346) at coursierapi.shaded.coursier.internal.api.ApiHelper.doFetch(ApiHelper.scala) at coursierapi.Fetch.fetchResult(Fetch.java:244) at coursierapi.Fetch.fetch(Fetch.java:239) at org.scalafmt.dynamic.CoursierDependencyDownloader.$anonfun$download$1(CoursierDependencyDownloader.scala:25) at scala.util.Try$.apply(Try.scala:210) at org.scalafmt.dynamic.CoursierDependencyDownloader.download(CoursierDependencyDownloader.scala:16) at org.scalafmt.dynamic.ScalafmtModuleLoader$WithDownloader.load(ScalafmtModuleLoader.scala:34) at org.scalafmt.dynamic.ScalafmtModuleLoader$CachedProxy.load$1(ScalafmtModuleLoader.scala:56) at org.scalafmt.dynamic.ScalafmtModuleLoader$CachedProxy.$anonfun$load$3(ScalafmtModuleLoader.scala:57) at scala.util.Try$.apply(Try.scala:210) at org.scalafmt.dynamic.utils.ReentrantCache.getOrAddToCache(ReentrantCache.scala:41) at org.scalafmt.dynamic.ScalafmtModuleLoader$CachedProxy.load(ScalafmtModuleLoader.scala:57) at org.scalafmt.dynamic.ScalafmtConfigLoader$.$anonfun$load$1(ScalafmtConfigLoader.scala:34) at scala.util.Either.flatMap(Either.scala:352) at org.scalafmt.dynamic.ScalafmtConfigLoader$.load(ScalafmtConfigLoader.scala:32) at org.scalafmt.dynamic.ScalafmtConfigLoader$CachedProxy.load$1(ScalafmtConfigLoader.scala:74) at org.scalafmt.dynamic.ScalafmtConfigLoader$CachedProxy.$anonfun$load$12(ScalafmtConfigLoader.scala:76) at scala.util.Try$.apply(Try.scala:210) at org.scalafmt.dynamic.utils.ReentrantCache.getOrAddToCache(ReentrantCache.scala:41) at org.scalafmt.dynamic.ScalafmtConfigLoader$CachedProxy.$anonfun$load$7(ScalafmtConfigLoader.scala:76) at scala.util.Success.fold(Try.scala:281) at org.scalafmt.dynamic.ScalafmtConfigLoader$CachedProxy.load(ScalafmtConfigLoader.scala:70) at org.scalafmt.dynamic.ScalafmtDynamic.resolveConfig(ScalafmtDynamic.scala:53) at org.scalafmt.dynamic.ScalafmtDynamic.createSession(ScalafmtDynamic.scala:47) at org.scalafmt.dynamic.ScalafmtDynamic.format(ScalafmtDynamic.scala:44) at com.softwaremill.adopttapir.starter.FormatScalaFiles$.$anonfun$apply$4(FormatScalaFiles.scala:22) at scala.collection.IterableOnceOps.foreach(IterableOnce.scala:563) at scala.collection.IterableOnceOps.foreach$(IterableOnce.scala:561) at scala.collection.AbstractIterator.foreach(Iterator.scala:1293) at com.softwaremill.adopttapir.starter.FormatScalaFiles$.$anonfun$apply$1(FormatScalaFiles.scala:20) at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.scala:18) at cats.effect.unsafe.WorkerThread.blockOn(WorkerThread.scala:591) at scala.concurrent.package$.blocking(package.scala:124) at cats.effect.IOFiber.runLoop(IOFiber.scala:967) at cats.effect.IOFiber.execR(IOFiber.scala:1332) at cats.effect.IOFiber.run(IOFiber.scala:139) at cats.effect.unsafe.WorkerThread.run(WorkerThread.scala:461)
```

# Walkaround
Restart of application helps temporarily.

# Proposed solution.

As this loader uses `coursier` for downloading all necessary `jars` the main idea is to provide them ahead of time. 
For that purpose we could attach volume with already fullfilled coursier cache which will be persisted and independent from deployment of an app. 

Proposed tasks:
- [ ]  create in docker mountable cache for coursier with write access
    - [ ] add persistent volume in k8s with read/write access with already attached required files. 
    `kubectl cp <pod>:/home/demiourgos728/.cache <local_destination>
- [ ]  setup readines probe to check zip endpoint

**Comment by katlasik** [comment]
It should be fixed with #280 .

**Comment by adamw** [comment]
Didn't happen since the fix, clsoing
